app/shop.py

```diff
@@ -0,0 +1,57 @@
+from typing import List, Dict, Union
+from datetime import datetime
+from unittest.mock import MagicMock
+
+from app.customer import Customer
+from app.trip import Trip
+
+
+class Shop:
+
+    def __init__(
+        self,
+        name: str,
+        location: List[int],
+        products: Dict[str, Union[int, float]]
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    @staticmethod
+    def convert_to_int_if_possible(number: float) -> float | int:
+        if number == int(number):
+            return int(number)
+        return number
+
+    @staticmethod
+    def shop_receipt(
+        customer: Customer,
+        trip_instances: List[Trip]
+    ) -> None:
+        cheapest_trip = Trip.cheapest_trip(trip_instances)
+        datetime_mock = MagicMock(wrap=datetime)
+        datetime_mock.now.return_value = datetime(2021, 1, 4, 12, 33, 41)
+
+        milk_cost = Shop.convert_to_int_if_possible(
```

Comment:
```suggestion
        milk_cost = self.convert_to_int_if_possible(
```

Reply:
it is better to operate at the level of class instances than at the level of the classes

Reply:
fix it wherever you can

Reply:
yes, that's much better) 
I fixed it where I could